### PR TITLE
gpu affinity

### PR DESCRIPTION
**Pull request by jmagoon**
- add gpu for cloud runners

### CODE REVIEW

File: translator.go [hunk 1]
```diff
@@ -239,6 +240,63 @@ func blockTemplate(block *zjson.Block, blockKey string, key string, organization
 			},
 		},
 	}
+
+	if block.Action.Resources.CPU.Request != "" {
+		template.Container.Resources.Requests[corev1.ResourceCPU] = resource.MustParse(block.Action.Resources.CPU.Request)
```
Comment (plparent) [inline translator.go:245]: I believe MustParse will panic if it fails. We should regular Parse and check if err is nil.

File: translator.go [hunk 2]
```diff
@@ -239,6 +240,63 @@ func blockTemplate(block *zjson.Block, blockKey string, key string, organization
...
+				Key:      "nvidia.com/gpu",
+				Operator: corev1.TolerationOpExists,
+				Effect:   corev1.TaintEffectNoSchedule,
+			},
+		}
+	} else {
+		// For non-GPU jobs, we don't need specific affinity or tolerations
+		// They will naturally schedule on the non-GPU nodes
+		template.Affinity = nil
+		template.Tolerations = nil
```
Comment (plparent) [inline translator.go:296]: Is this else statement needed at all?

**Review by plparent — CHANGES_REQUESTED** [review]

**Review by FGRCL — COMMENTED** [review]
image
It's small but, in dark mode,  `TextInput` value is darker than the placeholder.

**Comment by FGRCL** [comment]
Personally, I also think that I could use some indication on the valid units for the resources.

File: translator.go (hunk 1 above)
Comment (jmagoon) [reply translator.go]: Fixed

File: translator.go (hunk 2 above)
Comment (jmagoon) [reply translator.go]: Nope

**Pull request merged**